cinema/views.py

```diff
@@ -56,3 +91,24 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+        rows = "cinema_hall__rows"
+        seats = "cinema_hall__seats_in_row"
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+        queryset = queryset.select_related("cinema_hall").annotate(
+            tickets_available=F(rows) * F(seats) - Count("tickets")
+        ).order_by("id")
+        return queryset
```

Comment:
The `get_queryset` method in `MovieSessionViewSet` correctly uses annotations to calculate `tickets_available`. Ensure that this logic aligns with your application's requirements for calculating available tickets.